ddgr

```diff
@@ -1139,18 +1140,25 @@ class DdgCmd(object):
         parser = DdgParser()
         parser.feed(page.text)
 
-        self.results = parser.results
+        if self.options.pagination:
+            self.results.extend(parser.results)
+            offset = len(self._urltable)
+            for r in self.results:
```

Comment:
Are we doing this again and again for ALL results? Aren't the earlier results already indexed correctly?